x-pack/plugins/ml/public/application/contexts/ml/data_source_context.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, useCallback, useContext, useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useLocation } from 'react-router-dom';
+import { parse } from 'query-string';
+import { i18n } from '@kbn/i18n';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { SavedSearch } from '@kbn/saved-search-plugin/public';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { SavedSearchSavedObject } from '../../../../common/types/kibana';
+import { DataViewAndSavedSearch } from '../../util/index_utils';
+import { useMlKibana } from '../kibana';
+import { createSearchItems } from '../../jobs/new_job/utils/new_job_utils';
+
+export interface DataSourceContextValue {
+  combinedQuery: any;
+  currentDataView: DataView; // TODO this should be DataView or null
+  // @deprecated currentSavedSearch is of SavedSearchSavedObject type, change to selectedSavedSearch
+  deprecatedSavedSearchObj: SavedSearchSavedObject | null;
+  selectedSavedSearch: SavedSearch | null;
+}
+
+export const DataSourceContext = React.createContext<DataSourceContextValue>(
+  {} as DataSourceContextValue
+);
+
+/**
+ * Context provider that resolver current data view and the saved search from the URL state.
```

Comment:
Nit - should this be `that resolves ...` rather than `that resolver` ?

Reply:
fixed in https://github.com/elastic/kibana/pull/159176/commits/03b083702603bfd42284d0be20fefbb06622d41f